compiler/resolution/functionResolution.cpp

```diff
@@ -10596,9 +10596,23 @@ Expr* resolveExpr(Expr* expr) {
   } else if (SymExpr* se = toSymExpr(expr)) {
     makeRefType(se->symbol()->type);
 
-    if (ForallStmt* pfs = isForallIterExpr(se)) {
+    ForallStmt* pfs = isForallIterExpr(se);
+    ForLoop* pfl = toForLoop(se->parentExpr);
```

Comment:
Creating both pfs/pfl feels redundant again.

Reply:
In this case we really do, do different things depending on if its a foreach or forall loop. For forall loops we call `resolveForallHeader` which does some extra stuff in addition to `setupAndResolveShadowVars` (which we also do for `foreach` loops).

In the long run I might end up abstracting `resolveForallHeader` to work with `foreach` loops as well but I won't tackle that this PR.

So I think we still need both branches.

As far as doing something like `else if(ForLoop* pfl = ...) { ... }` instead of calculating `pfl` above, in this case since there's more to that condition I want to `&&` together that needs that pointer `pfl` to be assigned ahead of time. I could duplicate `toForLoop(se->parentExpr)` multiple times (uggh) or I could nest it but this is part of a larger chain of "if/else if" blocks so that's not easily doable either.

I'm going to factor this logic out into its own function and then call that (so atleast it looks less ugly at this site).